C error using fixed length array and inline multiple assign with self values

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
OS: macos, macOS, 12.5.1, 21G83
Processor: 8 cpus, 64bit, little endian, Apple M1
CC version: Apple clang version 13.1.6 (clang-1316.0.21.2.5)

getwd: /Users/samuel/coding/basics
vmodules: /Users/samuel/.vmodules
vroot: /Users/samuel/v
vexe: /Users/samuel/v/v
vexe mtime: 2022-08-28 07:02:41
is vroot writable: true
is vmodules writable: true
V full version: V 0.3.0 06e153d.797bdd5

Git version: git version 2.32.1 (Apple Git-133)
Git vroot status: weekly.2022.34-40-g797bdd5e
.git/config present: true
thirdparty/tcc status: thirdparty-macos-arm64 173c526e

```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg fibonacci.v`
```v
fn main() {
	number := 5
	ans := fib(number)

	println(ans)
}

fn fib(n int) u64 {
	if n <= 0 { panic("Bad number") }
	return match n {
		1 | 2 { 1 }
		else { 
			mut pair := [1,2]!
			for _ in 0..n {
				pair[0], pair[1] = pair[1], pair[0] + pair[1]
			}
			pair[1]
		}
	}

}
```

**What did you expect to see?**

21


**What did you see instead?**
```
==================
   ^~~~~~~~~~~~~~~
/tmp/v_501/fibonacci.3624317353655210864.tmp.c:12022:14: error: use of undeclared identifier '_var_218'
                        pair[0] = _var_218;
                                  ^
/tmp/v_501/fibonacci.3624317353655210864.tmp.c:12023:14: error: use of undeclared identifier '_var_209'
                        pair[1] = _var_209+_var_218;
                                  ^
/tmp/v_501/fibonacci.3624317353655210864.tmp.c:12023:23: error: use of undeclared identifier '_var_218'
                        pair[1] = _var_209+_var_218;
                                           ^
1 warning and 3 errors generated.
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

--- a/vlib/v/gen/c/assign.v
+++ b/vlib/v/gen/c/assign.v
@@ -658,6 +658,30 @@ fn (mut g Gen) gen_cross_var_assign(node &ast.AssignStmt) {
 					g.write(', ')
 					g.expr(left.index)
 					g.writeln(');')
+				} else if sym.kind == .array_fixed {
+					info := sym.info as ast.ArrayFixed
+					elem_typ := g.table.sym(info.elem_type)
+					if elem_typ.kind == .function {
+						left_typ := node.left_types[i]
+						left_sym := g.table.sym(left_typ)
+						g.write_fn_ptr_decl(left_sym.info as ast.FnType, '_var_$left.pos.pos')
+						g.write(' = *(voidptr*)')
+					} else {
+						styp := g.typ(info.elem_type)
+						g.write('$styp _var_$left.pos.pos = ')
+					}
+					if left.left_type.is_ptr() {
+						g.write('*')
+					}
+					needs_clone := info.elem_type == ast.string_type && g.is_autofree
+					if needs_clone {
+						g.write('/*1*/string_clone(')
+					}
+					g.expr(left)
+					if needs_clone {
+						g.write(')')
+					}
+					g.writeln(';')
 				} else if sym.kind == .map {
 					info := sym.info as ast.Map
 					skeytyp := g.typ(info.key_type)

--- a/vlib/v/tests/cross_assign_fixed_array_test.v
+++ b/vlib/v/tests/cross_assign_fixed_array_test.v
@@ -1,0 +1,25 @@
+fn test_cross_assign_fixed_array() {
+	number := 5
+	ans := fib(number)
+
+	println(ans)
+	assert ans == 21
+}
+
+fn fib(n int) u64 {
+	if n <= 0 {
+		panic('Bad number')
+	}
+	return match n {
+		1 | 2 {
+			1
+		}
+		else {
+			mut pair := [1, 2]!
+			for _ in 0 .. n {
+				pair[0], pair[1] = pair[1], pair[0] + pair[1]
+			}
+			pair[1]
+		}
+	}
+}

